span_implementation.go

```diff
@@ -25,9 +25,10 @@ import (
 type spanImpl struct {
 	mtx sync.RWMutex
 	model.SpanModel
-	tracer        *Tracer
-	mustCollect   int32 // used as atomic bool (1 = true, 0 = false)
-	flushOnFinish bool
+	tracer              *Tracer
+	mustCollect         int32 // used as atomic bool (1 = true, 0 = false)
+	flushOnFinish       bool
+	finishedSpanHandler func(*model.SpanModel) bool
```

Comment:
fyi we have a list of these in java (ex one for metrics, one for dependency graph)